sling/core/console/src/main/java/com/composum/sling/nodes/servlet/SourceModel.java

```diff
@@ -178,12 +178,7 @@ public String getName() {
     }
 
     public String getPrimaryType() {
-        try {
-            return resource.adaptTo(Node.class).getPrimaryNodeType().getName();
-        } catch (RepositoryException rex) {
-            LOG.error(rex.getMessage(), rex);
-        }
-        return "";
+        return ResourceUtil.getPrimaryType(resource);
```

Comment:
this returns probably &lt;null&gt; instead of "" as before with a possibly NPE in the SourceModel